### ISSUE TITLE
Video retains full-screen resolution after exiting full-screen

### ISSUE THREAD

**Issue by OSLDNKKfEUVTABDLiNgk** [mpv-player/mpv#9406]
### Important Information
- mpv version 0.34
- EndeavourOS; Build ID: 2021.08.27 (archlinux)
- ArchLinux repositories
- Problem was present in .33 and is present in .34.
- mutter 40.5
- amdgpu (Radeon RX Vega (VEGA10, DRM 3.42.0, 5.14.16-zen1-1-zen, LLVM 12.0.1))

### Reproduction steps
Bug still present with --no-config. Set video to full-screen, then exit full-screen.
### Expected behavior
Video returns to its previous scale. EG a `1920*1080` video would return to a `1920*1080` window after exiting full-screen mode
### Actual behavior
Video retains 2 scaling. `1920*1080` video displays as a `3840*2160` video
### Log file
[mpv.log](https://github.com/mpv-player/mpv/files/7491396/mpv.log)
### Sample files
Reproducible with any video - test video is copyright encumbered.

**Comment by Dudemanguy** [comment]
Hmm you're right. I haven't pulled up GNOME in a while but this used to not happen. Workaround: if you pause it before switching fullscreen it doesn't occur.

**Comment by OSLDNKKfEUVTABDLiNgk** [comment]
Bug is not present on another PC
mpv .32
Debian 11
Debian repositories
i915 driver running on Intel T9600 (GM45) integrated graphics.
GNOME 3.38.5

**Comment by OSLDNKKfEUVTABDLiNgk** [comment]
If you toggle full-screen while paused the video returns to 1 scaling sometimes. More frequently it retains the 2.0 scaling resolution.

**Comment by Dudemanguy** [comment]
Here's what I observed:

```
[   2.877][v][vo/gpu/wayland] Resizing due to xdg from 3840x2160 to 1920x1080
[   2.879][v][vo/gpu/wayland] Handling resize on the egl side
[   2.879][v][vo/gpu] Resize: 1920x1080
[   2.879][v][vo/gpu] Window size: 1920x1080 (Borders: l=0 t=0 r=0 b=0)
[   2.879][v][vo/gpu] Video source: 1920x1080 (1:1)
[   2.879][v][vo/gpu] Video display: (0, 0) 1920x1080 -> (0, 0) 1920x1080
[   2.879][v][vo/gpu] Video scale: 1.000000/1.000000
[   2.879][v][vo/gpu] OSD borders: l=0 t=0 r=0 b=0
[   2.879][v][vo/gpu] Video borders: l=0 t=0 r=0 b=0
[   2.879][v][vo/gpu] Reported display depth: 10
[   2.879][d][osc] osc_init 
[   2.879][d][vo/gpu] Resizing texture: 960x540
[   2.880][d][cplayer] Run command: change-list, flags=64, args=[name="shared-script-properties", operation="append", value="osc-margins=0.000000,0.000000,0.000000,0.000000"]
[   2.880][v][cplayer] Set property: shared-script-properties -> 1
[   2.880][d][cplayer] Run command: disable-section, flags=64, args=[name="window-controls"]
[   2.880][d][cplayer] Run command: disable-section, flags=64, args=[name="window-controls"]
[   2.881][d][vo/gpu] Resizing texture: 960x1080
[   2.884][d][vo/gpu] Resizing texture: 1920x1080
[   2.887][d][vo/gpu] Resizing texture: 1920x1080
[   2.887][d][vo/gpu] Resizing texture: 1920x1080
[   2.889][v][vo/gpu/wayland] Resizing due to xdg from 1920x1080 to 3840x2160
[   2.901][v][vo/gpu/wayland] Handling resize on the egl side
[   2.901][v][vo/gpu] Resize: 3840x2160
```

The first line is taking it out of full screen. The "Resizing due to xdg from 1920x1080 to 3840x2160" isn't triggered by the user (aka I didn't do anything) so that means it's mutter sending us a toplevel config event and mpv is handling it. It seems like mutter is sending an erroneous event (perhaps some compositor race between mpv painting frames at fullscreen and the resize) so I think this is their bug.

**Comment by Dudemanguy** [comment]
Seems like it is fixed in mutter 41.1. Going to go ahead and close this now. Feel free to reopen if that is not the case for you.

**Comment by OSLDNKKfEUVTABDLiNgk** [comment]
Just updated to Gnome 41 and the bug is gone, you're right.
